database/migrations/2021_03_06_201232_add_url_to_achievement_table.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddUrlToAchievementTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('achievement', function(Blueprint $table) {
+            $table->boolean('has_page')->after('tier');
+            $table->string('page_name')->nullable()->after('has_page');
```

Comment:
If this is used as url, it should be unique. Next to that you'll have to think about url encoding and maybe a seperate name and url field where the url field is highly constrained to a set of allowed characters.